ix-dev/test/traefik/templates/docker-compose.yaml

```diff
@@ -0,0 +1,121 @@
+{% set traefik = tpl.add_container(values.consts.traefik_container_name, "image") %}
+
+{# Configure Healthcheck #}
+{% do traefik.healthcheck.set_test("http", {"port": values.network.web_port.port_number}) %}
+
+{# Add capabilities #}
+{% do traefik.add_caps(["SETUID", "SETGID", "CHOWN", "FOWNER", "DAC_OVERRIDE"]) %}
+
+{# Add Commands to a variable as the set.command argument can only be used one time. #}
+{% set commands = namespace(x=[
+  "--ping=true",
+  "--api=%s" | format(values.consts.commands.api.enabled),
+  "--api.dashboard=%s" | format(values.consts.commands.api.dashboard),
+  "--api.insecure=%s" | format(values.consts.commands.api.insecure),
+  "--api.disabledashboardad=true",
+  "--providers.docker=%s" | format(values.consts.commands.docker.enabled),
+  "--providers.docker.exposedbydefault=%s" | format(values.consts.commands.docker.exposed),
+  "--providers.docker.network=%s" | format(values.consts.commands.docker.proxy_network),
+]) %}
+{# Add conditional commands
+   EntryPoints
+   Check if http proxying is enabled and add entrypoint when nessecary #}
+{% if values.network.http_enable %}
+  {% do commands.x.append("--entryPoints.web.address=:%s" | format(values.network.http_port.port_number)) %}
+{% endif %}
+{#Check if https proxying is enabled and add entrypoint when nessecary #}
+{% if values.network.https_enable %}
+  {% do commands.x.append("--entryPoints.websecure.address=:%s" | format(values.network.https_port.port_number)) %}
+{% endif %}
+{#Check if https redirection is enabled and add enable https redirection if it is #}
+{% if values.network.http_enable and values.network.https_enable and values.network.https_redirect %}
+  {% do commands.x.append("--entrypoints.web.http.redirections.entrypoint.to=websecure") %}
+  {% do commands.x.append("--entrypoints.web.http.redirections.entrypoint.scheme=https") %}
+  {% do commands.x.append("--entrypoints.web.http.redirections.entrypoint.permanent=true") %}
+{% endif %}
+
+{# Setup Proxy commands
+   HTTP Challenge #}
+{% if values.proxy.challenge == "http_challenge" %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.httpchallenge=true") %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.httpchallenge.entrypoint=web") %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.email=%s" | format(values.proxy.email)) %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.storage=/letsencrypt/acme.json") %}
+{% endif %}
+{# TLS Challenge #}
+{% if values.proxy.challenge == "tls_challenge" %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.tlschallenge=true") %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.email=%s" | format(values.proxy.email)) %}
+  {% do commands.x.append("--certificatesresolvers.truenasresolver.acme.storage=/letsencrypt/acme.json") %}
+{% endif %}
+
+{# DNS Challenge BUILD AT SOME POINT (one provider at a time) #}
```

Comment:
nope, there wont be such thing. providers is an infinite increasing list.
yes I know that ddns-update does simillar thing, but that is because it replaced a system service for ddns.